### ISSUE TITLE
Blank white screen on Windows 10

### ISSUE THREAD

**Issue by WebDev222** [BKSalman/ytdlp-gui#22]
Hello! 
After opening the ytdlp-gui.exe file, an empty window opens for a few seconds and closes. Nothing else happens. I tried both as administrator and without. What can this problem be related to?

![изображение](https://github.com/BKSalman/ytdlp-gui/assets/123309880/cfda0f17-1a6f-4257-b199-72f9adab7e18)

**Comment by BKSalman** [comment]
Привет! :) (literally the only useful word I know lol)

first of all, could you run the application from powershell like this `ytdlp-gui.exe --version`?
if you are on the latest version it will print out the version and hopefully the git hash, which would make debugging easier.

also try running it from powershell with the `YTG_LOG` environment variable set like this:
```
$env:YTG_LOG=debug

ytdlp-gui.exe
```
and check if there is anything suspicious going on there probably red `ERROR` lines or something, and share it here

I'm not totally sure if that works because I haven't used powershell that much

Note: please don't run random applications as administrator unless you know what they do exactly.

**Comment by WebDev222** [comment]
When running the `ytdlp-gui.exe --version` command through PowerShell, nothing happens
![изображение](https://github.com/BKSalman/ytdlp-gui/assets/123309880/f907f6f4-2dc5-40fa-bcc4-fbc75e289467)

When the command 
```
$env:YTG_LOG=debug 
ytdlp-gui.exe 
```
is executed, the white ytdlp-gui.exe window opens and closes only for a few seconds. PowerShell says that "the name "debug" is not recognized as the name of a cmdlet, function, script file or executable"
![изображение](https://github.com/BKSalman/ytdlp-gui/assets/123309880/560f03cd-55dc-4797-9e02-06350ffe9d5d)

**Comment by BKSalman** [comment]
> When running the ytdlp-gui.exe --version command through PowerShell, nothing happens

this probably means you're not running the latest version `1.0.0`, so to start, you can download that from the [releases page](https://github.com/BKSalman/ytdlp-gui/releases/tag/v1.0.0)


> is executed, the white ytdlp-gui.exe window opens and closes only for a few seconds. PowerShell says that "the name "debug" is not recognized as the name of a cmdlet, function, script file or executable"

ah that's probably my mistake, I think you need to add quotes like this:
```
$env:YTG_LOG="debug"
```
then just run the application as normal
```
ytdlp-gui.exe
```

**Comment by WebDev222** [comment]
I didn't have any other versions, but just in case I downloaded the "ytdlp-gui-windows-64.zip" version again from the [releases page](https://github.com/BKSalman/ytdlp-gui/releases/tag/v1.0.0).
The result is almost the same: 
- when running the command `ytdlp-gui.exe --version` nothing happens;
- when running the command `$env:YTG_LOG="debug" ytdlp-gui.exe` a white program window opens for a few seconds, nothing is displayed in the terminal.

![изображение](https://github.com/BKSalman/ytdlp-gui/assets/123309880/9f539f29-d08c-49af-bd1d-a9e370f6954c)

**Comment by BKSalman** [comment]
this is very strange, I honestly don't know why this is not working, however you can check the log files in something like `C:\Users\Alice\AppData\Local\ytdlp-gui\logs` (not sure since I don't use windows now)

if you find it, and see something there that could be useful, please share it.

**Comment by BKSalman** [comment]
if all fails, you could try to [build the program from source](https://github.com/BKSalman/ytdlp-gui#build-from-source) and move it from `<clone-repository-root-folder>\target\release\ytdlp-gui` to the `C:\Users\ds\Documents\Soft\ytdlp-gui-windows-64` folder (because you need `ffmpeg` and `yt-dlp` to be in the same place since I didn't package it properly for windows), and see if the blank screen issue is solved

**Comment by WebDev222** [comment]
These are the logs I found at C:\Users\Alice\AppData\Local\ytdlp-gui\logs
[logs.zip](https://github.com/BKSalman/ytdlp-gui/files/12780389/logs.zip)

**Comment by BKSalman** [comment]
this is probably some graphics/rendering error, I have no idea how to solve it tbh, but if you want to try compiling the project from source it _might_ help

**Comment by WebDev222** [comment]
I compiled the program following these instructions, but the executables at `<project-root>/target/debug/ytdlp-gui` and `<project-root>/target/release/ytdlp-gui` do not have the extension `.exe`. If you add the `.exe` extension manually, the program does not start, the following message appears: "Unable to run this application on your PC. To find a version for your computer, please contact the publisher of the application".

![explorer_ccyTR3ihh0](https://github.com/BKSalman/ytdlp-gui/assets/123309880/13198204-e887-49be-a6ae-19f6f0a153d6)

Note: perhaps this is due to the fact that I compiled in Linux?

**Comment by BKSalman** [comment]
Indeed if you compile the binary on linux it's by default gonna compile it as an ELF (the executable format linux uses), and it also doesn't have an extension like windows .exe, you can either cross compile it on linux for windows, or just compile it on windows

**Comment by WebDev222** [comment]
Unfortunately, no reliable information about cross-compiling could be found. Could you please tell me how to implement it? Should I use `mingw-w64`?

**Comment by BKSalman** [comment]
This stackoverflow answer here has enough information I think

https://stackoverflow.com/a/62853319

tldr: 
- get mingw-w64 from your distribution's repository
- `rustup target add x86_64-pc-windows-gnu`
- `cargo build --target x86_64-pc-windows-gnu`

I kinda doubt that compiling it yourself would make any difference, but I hope it works

I think the main problem is with graphics and stuff, if the machine you're running the application on is old or very low end, that might be the issue, not that the application is heavy, but graphical drivers support and whatever (I'm not too knowledgeable about graphics)

**Comment by BKSalman** [comment]
ACTUALLY, you could try enabling the iced glow renderer by using this patch, then recompile it:

```
diff --git a/Cargo.toml b/Cargo.toml
index dc23203..2ea965e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,7 @@ members = [
 [dependencies]
 chrono = { version = "0.4.23", default-features = false, features = ["clock"] }
 dirs = "4.0.0"
-iced = { version = "0.9.0" }
+iced = { version = "0.9.0", features = ["glow"] }
 iced_aw = { version = "0.5.0", features = ["tabs", "modal", "card"], default-features = false }
 iced_native = "0.10.0"
 native-dialog = "0.6.3"
```

to apply the patch you can create a file in the repository, and put the text above inside of it, then run `git apply the-name-of-the-file`

sorry for the inconvenience since I don't really use windows, but we can try fixing it :)

if that doesn't work, maybe when I update iced (the GUI library used for this project) it could work, because it has a software renderer that probably supports more older hardware and stuff

**Comment by BKSalman** [comment]
Any update on this issue?

I understand if it was too much to ask you to do all of these things

**Comment by BKSalman** [comment]
since there was no reply I'm going to close this issue, if you still want help with this issue you are welcome to reopen it
